metadataserver/server_test.go

```diff
@@ -633,3 +635,69 @@ func TestTaskIdentityWithECDSA(t *testing.T) {
 		}
 	play(t, ess, tapes)
 }
+
+func TestRequireToken(t *testing.T) {
+	ess, err := setupStubServer(t)
+	if err != nil {
+		t.Fatal("Could not get stub server: ", err)
+	}
+
+	ms := setupMetadataServer(t, ess, ecdsaCerts[0], true)
+
+	// Get Token
+	tokenPath := fmt.Sprintf("http://%s%s", ess.proxyListener.Addr().String(), "/latest/api/token")
```

Comment:
Any reason we don't use the "official" ec2metadata client here? and just override the EC2 metdata URL?

Reply:
Good point. I’ll probably add that as another test since you can’t really inspect the response status codes directly.